src/terminal-screen.c

```diff
@@ -1949,6 +1949,8 @@ terminal_screen_child_exited (VteTerminal *terminal, int status)
 	switch (action)
 	{
 	case TERMINAL_EXIT_CLOSE:
+		if ((status == 9) && (priv->override_command == NULL))
+			break;
 		g_signal_emit (screen, signals[CLOSE_SCREEN], 0);
 		break;
```

Comment:
Apply the [De Morgan's law](https://en.wikipedia.org/wiki/De_Morgan%27s_laws): the negation of a conjunction is the disjunction of the negations:
```C
if ((status != 9) || (priv->override_command != NULL))
    g_signal_emit (screen, signals[CLOSE_SCREEN], 0);
break;
```

Reply:
yes, you are right, done, and the same change can be done inside `case TERMINAL_EXIT_HOLD`